src/mad_spark_alt/core/timeout_wrapper.py

```
@@ -0,0 +1,170 @@
+            progress_task = asyncio.create_task(_log_progress(phase, timeout))
+        
+        # Execute with timeout
+        result: Any = await asyncio.wait_for(coro(), timeout=timeout)
+        
+        # Cancel progress logging
+        if log_progress:
+            progress_task.cancel()
+            try:
+                await progress_task
+            except asyncio.CancelledError:
+                pass
```

Ruff (0.11.9)

Use `contextlib.suppress(asyncio.CancelledError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(asyncio.CancelledError)`

(SIM105)